src/status_im/ui/screens/home/views.cljs

```diff
@@ -87,15 +91,165 @@
+                                                            :margin-right 2}}]
+          [react/text-input {:placeholder     (i18n/label :t/search)
+                             :blur-on-submit  true
+                             :multiline       false
+                             :number-of-lines 1
+                             :ref             #(reset! input-ref %)
+                             :style           {:flex        1
+                                               :margin      0
+                                               :padding     0
+                                               :line-height 22
+                                               :font-size   15}
+                             :default-value   search-filter
+                             :on-focus
+                             #(do
+                                (when on-focus
+                                  (on-focus))
+                                (reset! input-is-focused? true))
+                             :on-change
+                             (fn [e]
+                               (let [native-event (.-nativeEvent e)
+                                     text         (.-text native-event)]
+                                 (when on-change
+                                   (on-change text))))}]]
+         (when show-cancel?
+           [react/touchable-highlight
+            {:on-press #(do
+                          (when on-cancel
+                            (on-cancel))
+                          (.blur @input-ref)
+                          (reset! input-is-focused? false))
+             :style {:margin-left 16}}
+            [react/text {:style {:color     colors/blue
+                                 :font-size 15}}
+             (i18n/label :t/cancel)]])]))))
+
+(defn home-empty-view []
+  [react/view styles/no-chats
+   [react/i18n-text {:style styles/no-chats-text :key :no-recent-chats}]])
+
+(defn home-filtered-items-list [chats browsers]
+  [list/section-list
+   {:sections [{:title :t/chats
+                :data chats}
+               {:title :t/browsers
+                :data browsers}
+               {:title :t/messages
+                :data []}]
+    :key-fn first
+    :render-section-header-fn (fn [{:keys [title data]}]
+                                [react/text {:style {:font-size   15
+                                                     :margin-left 16
+                                                     :margin-top  16
+                                                     :color       colors/gray}}
+                                 (i18n/label title)])
+    :render-section-footer-fn (fn [{:keys [title data]}]
+                                (when (empty? data)
+                                  (case title
+                                    "messages"
+                                    [list/big-list-item {:text          (i18n/label :t/messages-search-coming-soon)
```

Comment:
Seems that all these entries are identical except for the icon keyword and text label in the first element, so would be better to move `case` expression so that it applies to `[:text :icon]` keys only.